Simplified client and uploader constructors

DavidSubiros opened the pull request:
### What

- Simplified client and uploader constructors, removing `hasUserPsk` bool param
- Now a client/uploader always contains a plain client and a `user-defined psk` client.
- Upgraded dependencies
- Refactored errors

### How to review

- Make sure changes make sense
- Make sure unit tests pass
- (info) I tested the uploader by using it in the Cantabular CSV exporter.

### Who can review

anyone

DavidSubiros committed:
simiplified s3 and uploader constructors: now they always have an skd and a crypto client, sharing the session. Improved errors according to new standard

DavidSubiros committed:
fixed tests and Makefile

DavidSubiros committed:
fixed conflicts

DavidSubiros committed:
modified README according to new Constructors

redhug1 commented on client.go:
```diff
@@ -418,9 +418,7 @@ func (cli *S3) ValidateBucket() error {
-		return NewError(fmt.Errorf("validation error for bucket: %w", err), log.Data{
-			"bucket_name": cli.bucketName,
-		})
+		return err
```
I think the code in this function could be further optimised to not need an 'if' and just return the 'err' and thus be put back to the original before this feature branch ?

redhug1 reviewed (changes requested):
A few little things to do ...

redhug1 reviewed (changes requested):
Please can you also fix the following:
url.go, Variable 'url' collides with imported package name in lines: 50, 53, 59, 62, 65, 92, 114, 148, 179, 199
-=-=-
Should there be some test code that achieves coverage for NewUploader() in uploader.go ?
-=-=-
healthcheck.go, Unhandled error in lines 29, 33, 44, 47  ... should anything be done with the error returns, or just put them into a '_' discard receptor ?  What do other api's do ?
-=-=-
client_test.go:
Method call 'err.Error()' might lead to a nil pointer dereference, lines: 250, 455, 481, 516, 582
Unhandled error, lines: 44, 45, 60, 61, 76, 77, 154, 155, 171, 172

DavidSubiros commented:
- fixed url collisions
- I didn't cover the constructors in testing, to prevent having to create real sessions (I guess we can create an interface for the AWS session, but I thought the constructor code is ver small)
- Fixed unhandled errors. After thinking a bit about this, I think it makes sense to propagate the errors returned by state.Update(). The other errors are in fact expected cases being handled (critical health status), but the state update would be an unexpected error while trying to do something we assume will work.
- Fixed the client_test error handling by creating a new aux func and asserting errors.

DavidSubiros committed:
extended README with encryption examples, fixed name collision and error checking

redhug1 reviewed (approved):
Great work.
Thank you for addressing all those other non related problems found.
